### ISSUE TITLE
max-statements-per-line and es6 export

### ISSUE THREAD

**Issue by daltones** [eslint/eslint#6264]
**What version of ESLint are you using?**
v2.11.0

**What parser (default, Babel-ESLint, etc.) are you using?**
Default (babel-eslint shows the same issue)

**Please show your full configuration:**
```json
{
  "parserOptions": {
    "ecmaVersion": 6,
    "sourceType": "module"
  },
  "rules": {
    "max-statements-per-line": [1, {"max": 1}]
  }
}
```

**What did you do? Please include the actual source code causing the issue.**
```js
export default function someFunction() {
    return true;
}
```

**What did you expect to happen?**
No warnings.

**What actually happened? Please include the actual, raw output from ESLint.**
```

/path-to-the-file/index.js
  1:16  warning  This line has too many statements. Maximum allowed is 1  max-statements-per-line

✖ 1 problem (0 errors, 1 warning)

```

So… Maybe we could say that I have **two** statements because I'm declaring a function and exporting it. But in my opinion that is not the case, it's a special one related to ES6 syntax.

But, if we really consider that there are two statements in that line, it would be nice to have an option for `max-statements-per-line` rule that we could ignore cases like this.

**Comment by platinumazure** [comment]
Seems like a bug to me. My guess is we use FunctionDeclaration node as a "top level statement node", but in reality it is a child of the export node. We need to add a check that the FunctionDeclaration isn't a child of the export declaration node.

Or something like that. :grin:

**Comment by mysticatea** [comment]
Thank you for this issue.

I confirmed it.
This bug was introduced in #6192.
This rule should ignore child node of ExportDeclarations as similar to such IfStatement.

I'm working on this.

**Comment by mgtitimoli** [comment]
In case it helps, this is also happening when exporting classes as follows

```javascript
export default class MyClass {
    // ...
}
```

**Comment by daltones** [comment]
Thanks for the fast replies! :)
